napari/utils/events/_tests/test_event_migrations.py

```diff
@@ -0,0 +1,16 @@
+import pytest
+
+from napari.utils.events.migrations import deprecation_warning_event
+
+
+def test_deprecation_warning_event() -> None:
+    event = deprecation_warning_event(
+        "obj.events", "old", "new", "0.1.0", "0.0.0"
+    )
+
+    def _print(msg: str) -> None:
+        print(msg)
+
+    with pytest.warns(FutureWarning):
+        event.connect(_print)
+        event(msg="test")
```

Comment:
We still expect the event to be emitted right? Would be nice to update the `_print` function to do something easily assertable like modifying a dummy object or something?

Reply:
You're right, I added a dummy `Counter` class.